CONTRIBUTING.md

```diff
@@ -338,6 +348,45 @@ This involves the following steps:
    ```bash
    >>> git push origin --delete feature-add-new-important-headers
    ```
+
+## Code review
+Generally, pull requests should be of a "digestible" size so that any issues arising during a code review can be resolved within a few iterations. Interact with the code reviewer by whichever means is easiest for you, but use the "comments" feature on the Github webpages in the first instance so there's a record of the issues raised. Sometimes it'll be easier to follow up via zoom or face-to-face discussions but in any case, the discussion should ultimately be marked as resolved, and the changes made should be transparent. "All done" is fine; more detail is required if suggestions were (jointly) deemed to be inappropriate. No need to over-engineer any of this -- small pull requests can just be merged in. However, the workflow described below should be sufficiently robust to deal even with "monster" pull requests.
+
+The general workflow for approving a pull request is as follows:
+
+1. **Reviewer:** Go to the "Files changed" tab for the pull request
+
+      ![](doc/README/files_changed_tab.png)
+
+   and inspect the changes made.
+
+ 1. **Reviewer:** Add comments/requests for change by leaving a comment near the relevant part of the code. Hover over the line number until the blue plus sign appears
+    ![](doc/README/plus_sign.png)
+
+    Click on it to make the comments window appear. Describe the issue and if it's something that you feel *must* be resolved, press the green "Start a review" button. (If it's just a comment, use "Add single comment" instead.)
+    ![](doc/README/comment_requesting_changes.png)
+    
+1. **Reviewer:** Keep going until you've reviewed all the changed files and left comments on all relevant issues, then finish the review by pressing the green "Finish your review" button at the very top of the page.
+      ![](doc/README/finish_your_review.png)
+
+1. **Reviewer:** Add a brief overall summary of the issues raised in the window that appears:
+      ![](doc/README/finish_review2.png)
+
+   If the changes are deemed essential, check the "Request changes" radio button before submitting the review.
+
+1. **Reviewee:** Work your way through the comments and either change the code as suggested (do this in your local cloned version of your forked repository, and then commit/push the changes to your remote forked repository; they will then automatically be included into the pull request), and leave a comment to say that you've done so. Alternatively, discuss/clarify the issue with the reviewer. 
```

Comment:
> then commit/push the changes to your remote forked repository

might be worth adding that the commits should be on the same branch the the PR was originally made from